src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
 public class WorkWithFile {
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String data = readFile(fromFileName);
+        String report = generateReport(data);
+        writeToFile(report, toFileName);
+    }
+
+    public String readFile(String fileName) {
+        StringBuilder builder = new StringBuilder();
+        try {
+            try (InputStream inputStream = new FileInputStream(fileName);
+                    BufferedReader bufferedReader = new BufferedReader(
+                            new InputStreamReader(inputStream))) {
+                while (bufferedReader.ready()) {
+                    builder.append(bufferedReader.readLine()).append(System.lineSeparator());
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file: " + fileName, e);
+        }
+        return builder.toString();
+    }
+
+    public String generateReport(String data) {
+        int supply = 0;
+        int buy = 0;
+        String[] forReport = data.split(System.lineSeparator());
+        for (String s : forReport) {
+            if (s.contains("supply")) {
+                supply += Integer.parseInt(s.replaceAll("[a-zA-Z\\p{Punct}]", ""));
+            }
+            if (s.contains("buy")) {
+                buy += Integer.parseInt(s.replaceAll("[a-zA-Z\\p{Punct}]", ""));
```

Comment:
Please use constants for all magic values including names of operations and patterns